core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -280,11 +302,22 @@ private[spark] class TaskSchedulerImpl(
       }
     }
     if (!launchedTask) {
-      taskSet.abortIfCompletelyBlacklisted(executorIdToHost.keys)
+      blacklistTracker.foreach(taskSet.abortIfCompletelyBlacklisted(executorsByHost, _))
     }
     return launchedTask
   }
 
+  private[scheduler] def areAllExecutorsBlacklisted(): Boolean = {
+    blacklistTracker match {
+      case Some(bl) =>
+        executorsByHost.forall { case (host, execs) =>
```

Comment:
I know this isn't your naming, but can you change this to hostToExecutors? I find that easier to read, and it's more consistent with map naming elsewhere.